test/integration/targets/ec2_instance_a/tasks/external_resource_attach.yml

```diff
@@ -115,27 +99,32 @@
           - "{{ presented_instance_fact.instances | length }} > 0"
           - "{{ checkmode_instance_fact.instances | length }} == 0"
 
-    - name: Terminate instance
-      ec2_instance:
-        filters:
-          tag:TestId: "{{ resource_prefix }}"
-        state: absent
-        wait: false
-        <<: *aws_connection_info
-      register: result
-
   always:
     - name: Terminate instance
       ec2_instance:
         instance_ids: "{{ in_test_vpc.instance_ids }}"
         state: absent
-        wait: false
-        <<: *aws_connection_info
+        # We can't delete the ENIs until we've deleted all the instances
+        wait: yes
       register: result
       ignore_errors: yes
 
     - ec2_key:
         name: "{{ resource_prefix }}_test_key"
         state: absent
-        <<: *aws_connection_info
+      ignore_errors: yes
+
+    - ec2_eni:
+        eni_id: '{{ eni_a.interface.id }}'
+        state: absent
+      ignore_errors: yes
```

Comment:
You could put these in a loop. Is delete_on_termination not working for the ec2_eni module? According to the documentation that may only work upon modification of ENIs.

Reply:
They didn't go away on termination when I was testing...